typeguard/__init__.py

```diff
@@ -632,7 +632,7 @@ def check_return_type(retval, memo: Optional[_CallMemo] = None) -> bool:
         try:
             check_type('the return value', retval, memo.type_hints['return'], memo)
         except TypeError as exc:  # suppress unnecessarily long tracebacks
-            raise exc from None
+            raise TypeError(exc) from None
```

Comment:
Use `**exc.args` as argument. Using the previous exception as argument makes the exception look like this: `TypeError(TypeError('argument "a" must be a callable',),)`